src/main/java/com/jcraft/jsch/OpenSshCertificateAwareIdentityFile.java

```diff
@@ -0,0 +1,285 @@
+ * </p>
+ */
+class OpenSshCertificateAwareIdentityFile implements Identity {
+
+  public static final String SSH_RSA_CERT_V01_AT_OPENSSH_DOT_COM = "ssh-rsa-cert-v01@openssh.com";
+  public static final String SSH_RSA_CERT = "ssh-rsa-cert";
+
+  public static final String SSH_DSS_CERT_V01_AT_OPENSSH_DOT_COM = "ssh-dss-cert-v01@openssh.com";
+  public static final String SSH_DSS_CERT = "ssh-dss-cert";
+
+  public static final String ECDSA_SHA2_NISTP256_CERT_V01_AT_OPENSSH_DOT_COM =
+      "ecdsa-sha2-nistp256-cert-v01@openssh.com";
+  public static final String ECDSA_SHA2_NISTP256_CERT = "ecdsa-sha2-nistp256-cert";
+
+  public static final String ECDSA_SHA2_NISTP384_CERT_V01_AT_OPENSSH_DOT_COM =
+      "ecdsa-sha2-nistp384-cert-v01@openssh.com";
+  public static final String ECDSA_SHA2_NISTP384_CERT = "ecdsa-sha2-nistp384-cert";
+
+  public static final String ECDSA_SHA2_NISTP521_CERT_V01_AT_OPENSSH_DOT_COM =
+      "ecdsa-sha2-nistp521-cert-v01@openssh.com";
+  public static final String ECDSA_SHA2_NISTP521_CERT = "ecdsa-sha2-nistp521-cert";
+
+
+  public static final String SSH_ED25519_CERT_V01_AT_OPENSSH_DOT_COM =
+      "ssh-ed25519-cert-v01@openssh.com";
+  public static final String SSH_ED25519_CERT = "ssh-ed25519-cert";
+
+  public static final String SSH_ED448_CERT_V01_AT_OPENSSH_DOT_COM =
+      "ssh-ed448-cert-v01@openssh.com";
+  public static final String SSH_ED448_CERT = "ssh-ed448-cert";
+
+
+  /**
+   * Determines if the given certificate file content as String, represents an OpenSSH certificate.
+   *
+   * @param certificateFileContent the certificate string to check
+   * @return {@code true} if the file content is a supported OpenSSH certificate type, {@code false}
+   *         otherwise
+   */
+  public static boolean isOpenSshCertificate(String certificateFileContent) {
+    String certificateKeyType = extractKeyType(trimToEmptyIfNull(certificateFileContent));
+    return isOpenSshCertificateKeyType(certificateKeyType);
+  }
+
+
+  /**
+   * Determines if the given public key type represents an OpenSSH certificate.
+   *
+   * @param publicKeyType the public key type string to check
+   * @return {@code true} if the type is a supported OpenSSH certificate type, {@code false}
+   *         otherwise
+   */
+  public static boolean isOpenSshCertificateKeyType(String publicKeyType) {
+    switch (publicKeyType) {
+      case SSH_RSA_CERT_V01_AT_OPENSSH_DOT_COM:
+      case SSH_DSS_CERT_V01_AT_OPENSSH_DOT_COM:
+      case ECDSA_SHA2_NISTP256_CERT_V01_AT_OPENSSH_DOT_COM:
+      case ECDSA_SHA2_NISTP384_CERT_V01_AT_OPENSSH_DOT_COM:
+      case ECDSA_SHA2_NISTP521_CERT_V01_AT_OPENSSH_DOT_COM:
+      case SSH_ED25519_CERT_V01_AT_OPENSSH_DOT_COM:
+      case SSH_ED448_CERT_V01_AT_OPENSSH_DOT_COM:
+      case SSH_RSA_CERT:
+      case SSH_DSS_CERT:
+      case ECDSA_SHA2_NISTP256_CERT:
+      case ECDSA_SHA2_NISTP384_CERT:
+      case ECDSA_SHA2_NISTP521_CERT:
+      case SSH_ED25519_CERT:
+      case SSH_ED448_CERT:
```

Comment:
Let's not support the non `@openssh.com` type certs for now, as we have no way of performing integration tests for them against an actual server implementation.
Once OpenSSH gets around to adding support for them, then we can add them.